diff -ruN a/third_party/proto/tensorflow/attr_value.proto b/third_party/proto/tensorflow/attr_value.proto
--- a/third_party/proto/tensorflow/attr_value.proto	2026-05-11 14:17:54.464313381 +0800
+++ b/third_party/proto/tensorflow/attr_value.proto	2026-05-11 14:17:54.468433540 +0800
@@ -2,9 +2,9 @@
 
 package tensorflow;
 
-import "tensorflow/core/framework/tensor.proto";
-import "tensorflow/core/framework/tensor_shape.proto";
-import "tensorflow/core/framework/types.proto";
+import "tensor.proto";
+import "tensor_shape.proto";
+import "types.proto";
 
 option cc_enable_arenas = true;
 option java_outer_classname = "AttrValueProtos";
diff -ruN a/third_party/proto/tensorflow/function.proto b/third_party/proto/tensorflow/function.proto
--- a/third_party/proto/tensorflow/function.proto	2026-05-11 14:17:54.464313381 +0800
+++ b/third_party/proto/tensorflow/function.proto	2026-05-11 14:17:54.468433540 +0800
@@ -2,9 +2,9 @@
 
 package tensorflow;
 
-import "tensorflow/core/framework/attr_value.proto";
-import "tensorflow/core/framework/node_def.proto";
-import "tensorflow/core/framework/op_def.proto";
+import "attr_value.proto";
+import "node_def.proto";
+import "op_def.proto";
 
 option cc_enable_arenas = true;
 option java_outer_classname = "FunctionProtos";
diff -ruN a/third_party/proto/tensorflow/graph.proto b/third_party/proto/tensorflow/graph.proto
--- a/third_party/proto/tensorflow/graph.proto	2026-05-11 14:17:54.464313381 +0800
+++ b/third_party/proto/tensorflow/graph.proto	2026-05-11 14:17:54.468433540 +0800
@@ -2,10 +2,10 @@
 
 package tensorflow;
 
-import "tensorflow/core/framework/function.proto";
-import "tensorflow/core/framework/graph_debug_info.proto";
-import "tensorflow/core/framework/node_def.proto";
-import "tensorflow/core/framework/versions.proto";
+import "function.proto";
+import "graph_debug_info.proto";
+import "node_def.proto";
+import "versions.proto";
 
 option cc_enable_arenas = true;
 option java_outer_classname = "GraphProtos";
diff -ruN a/third_party/proto/tensorflow/lite/schema.fbs b/third_party/proto/tensorflow/lite/schema.fbs
--- a/third_party/proto/tensorflow/lite/schema.fbs	2026-05-11 14:17:54.472553699 +0800
+++ b/third_party/proto/tensorflow/lite/schema.fbs	2026-05-11 14:17:54.472553699 +0800
@@ -17,8 +17,6 @@
 // Version 1: Add subgraphs to schema.
 // Version 2: Rename operators to conform to NN API.
 // Version 3: Move buffer data from Model.Subgraph.Tensors to Model.Buffers.
-// Version 3a: Add new builtin op code field. Has backward compatibility with
-//             version 3.
 // Version 3b: Rename fields in SignatureDef. Has backward compatibility with
 //             version 3 and 3a.
 // Version 3c: Move constant tensor buffers & custom op buffers outside from
@@ -48,7 +46,6 @@
   COMPLEX64 = 8,
   INT8 = 9,
   FLOAT64 = 10,
-  COMPLEX128 = 11,
   UINT64 = 12,
   // Experimental: Resource and variant types are experimental, that are subject
   // to change. Do not implement custom kernels using resource & variant types
@@ -963,8 +960,6 @@
 
 table AddOptions {
   fused_activation_function:ActivationFunctionType;
-  // Parameters supported by version 3.
-  pot_scale_int16:bool = true;
 }
 
 table MulOptions {
@@ -1092,8 +1087,6 @@
 
 table SubOptions {
   fused_activation_function:ActivationFunctionType;
-  // Parameters supported by version 5
-  pot_scale_int16:bool = true;
 }
 
 table DivOptions {
@@ -1619,35 +1612,6 @@
   buffer:uint;
 }
 
-// Map from an alias name of tensor to tensor index in the graph.
-// This is used in Signature def.
-table TensorMap {
-  // Represents the alias to use for this tensor.
-  name:string;
-
-  // The actual tensor index in the primary graph, that 'name' corresponds to.
-  tensor_index:uint;
-}
-
-// This corresponds to SignatureDef in Tensorflow SavedModel.
-// The SignatureDef will be part of the SavedModel provided for conversion.
-table SignatureDef {
-  // Named inputs for this signature.
-  inputs:[TensorMap];
-
-  // Named outputs for this signature.
-  outputs:[TensorMap];
-
-  // Key value which was in the Tensorflow SavedModel SignatureDef map.
-  signature_key:string;
-
-  // Model tag, deprecated.
-  deprecated_tag:string (deprecated);
-
-  // Index of subgraphs that corresponds to the exported method.
-  subgraph_index:uint;
-}
-
 table Model {
   // Version of the schema.
   version:uint;
@@ -1676,9 +1640,6 @@
 
   // Metadata about the model.
   metadata:[Metadata];
-
-  // Optional SignatureDefs for the model.
-  signature_defs:[SignatureDef];
 }
 
 root_type Model;
diff -ruN a/third_party/proto/tensorflow/node_def.proto b/third_party/proto/tensorflow/node_def.proto
--- a/third_party/proto/tensorflow/node_def.proto	2026-05-11 14:17:54.464313381 +0800
+++ b/third_party/proto/tensorflow/node_def.proto	2026-05-11 14:17:54.468433540 +0800
@@ -2,8 +2,8 @@
 
 package tensorflow;
 
-import "tensorflow/core/framework/attr_value.proto";
-import "tensorflow/core/framework/full_type.proto";
+import "attr_value.proto";
+import "full_type.proto";
 
 option cc_enable_arenas = true;
 option java_outer_classname = "NodeProto";
diff -ruN a/third_party/proto/tensorflow/op_def.proto b/third_party/proto/tensorflow/op_def.proto
--- a/third_party/proto/tensorflow/op_def.proto	2026-05-11 14:17:54.464313381 +0800
+++ b/third_party/proto/tensorflow/op_def.proto	2026-05-11 14:17:54.468433540 +0800
@@ -2,10 +2,10 @@
 
 package tensorflow;
 
-import "tensorflow/core/framework/attr_value.proto";
-import "tensorflow/core/framework/full_type.proto";
-import "tensorflow/core/framework/resource_handle.proto";
-import "tensorflow/core/framework/types.proto";
+import "attr_value.proto";
+import "full_type.proto";
+import "resource_handle.proto";
+import "types.proto";
 
 option cc_enable_arenas = true;
 option java_outer_classname = "OpDefProtos";
diff -ruN a/third_party/proto/tensorflow/resource_handle.proto b/third_party/proto/tensorflow/resource_handle.proto
--- a/third_party/proto/tensorflow/resource_handle.proto	2026-05-11 14:17:54.464313381 +0800
+++ b/third_party/proto/tensorflow/resource_handle.proto	2026-05-11 14:17:54.468433540 +0800
@@ -2,8 +2,8 @@
 
 package tensorflow;
 
-import "tensorflow/core/framework/tensor_shape.proto";
-import "tensorflow/core/framework/types.proto";
+import "tensor_shape.proto";
+import "types.proto";
 
 option cc_enable_arenas = true;
 option java_outer_classname = "ResourceHandle";
diff -ruN a/third_party/proto/tensorflow/tensor.proto b/third_party/proto/tensorflow/tensor.proto
--- a/third_party/proto/tensorflow/tensor.proto	2026-05-11 14:17:54.464313381 +0800
+++ b/third_party/proto/tensorflow/tensor.proto	2026-05-11 14:17:54.468433540 +0800
@@ -2,9 +2,9 @@
 
 package tensorflow;
 
-import "tensorflow/core/framework/resource_handle.proto";
-import "tensorflow/core/framework/tensor_shape.proto";
-import "tensorflow/core/framework/types.proto";
+import "resource_handle.proto";
+import "tensor_shape.proto";
+import "types.proto";
 
 option cc_enable_arenas = true;
 option java_outer_classname = "TensorProtos";
